File: src/neoforge/game/common/main/java/settingdust/preloading_tricks/neoforge/PreloadingTricksNeoForge.java
package settingdust.preloading_tricks.neoforge;

import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.common.Mod;
import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent;
import net.neoforged.fml.event.lifecycle.FMLCommonSetupEvent;
import settingdust.preloading_tricks.PreloadingTricks;
import settingdust.preloading_tricks.util.Entrypoint;

@Mod(PreloadingTricks.ID)
public class PreloadingTricksNeoForge {
    public PreloadingTricksNeoForge(IEventBus modEventBus) {
        @SuppressWarnings("unused")
        String id = PreloadingTricks.ID;
        Entrypoint.construct();

        modEventBus.addListener(this::commonSetup);
        modEventBus.addListener(this::clientSetup);
    }

    private void commonSetup(FMLCommonSetupEvent event) {
        Entrypoint.init();
    }

    private void clientSetup(FMLClientSetupEvent event) {
        Entrypoint.clientInit();
    }
}

File: src/neoforge/1.21/main/kotlin/settingdust/preloading_tricks/neoforge/PreloadingTricksNeoForge.kt
package settingdust.preloading_tricks.neoforge

import net.neoforged.fml.common.Mod
import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent
import net.neoforged.fml.event.lifecycle.FMLCommonSetupEvent
import settingdust.preloading_tricks.PreloadingTricks
import settingdust.preloading_tricks.util.Entrypoint
import thedarkcolour.kotlinforforge.neoforge.forge.MOD_BUS

@Mod(PreloadingTricks.ID)
object PreloadingTricksNeoForge {
    init {
        requireNotNull(PreloadingTricks)
        Entrypoint.construct()
        MOD_BUS.apply {
            addListener<FMLCommonSetupEvent> {
                Entrypoint.init()
            }
            addListener<FMLClientSetupEvent> { Entrypoint.clientInit() }
        }
    }
}